doc/languages-frameworks/rust.md

```diff
@@ -57,6 +57,85 @@ checksum can be then take from the failed build.
 To install crates with nix there is also an experimental project called
 [nixcrates](https://github.com/fractalide/nixcrates).
 
+## Compiling Rust crates using Nix instead of Cargo
+
+When run, `cargo build` produces a file called `Cargo.lock`,
+containing pinned versions of all dependencies. Nixpkgs contains a
+tool called `carnix` (`nix-env -iA nixos.carnix`), which can be used
+to turn a `Cargo.lock` into a Nix expression.
+
+That Nix expression calls `rustc` directly (hence bypassing Cargo),
+and can be used to compile a crate and all its dependencies. Here is
+an example for a minimal `hello` crate:
+
+
+    $ cargo new hello
+    $ cd hello
+    $ cargo build
+     Compiling hello v0.1.0 (file:///tmp/hello)
+      Finished dev [unoptimized + debuginfo] target(s) in 0.20 secs
+    $ carnix -o hello.nix --src ./. Cargo.lock
+    $ nix-build hello.nix
```

Comment:
`nix-build hello.nix` no longer works, since carnix replaced the `with import <nixpkgs>{};` preamble.

Also, this command would build all crates in `hello.nix` which might lead to errors if, for example, a crate like *redox_syscall* is pulled in as a conditional dependency of the *time* crate for the Redox-OS platform when building on Linux or any platform that is not Redox-OS.

i.e.
```
  redox_syscall_0_1_18 = redox_syscall_0_1_18_ {};
  time_0_1_37 = time_0_1_37_ {
    dependencies = [ libc_0_2_24 ]
      ++ (if buildPlatform.parsed.kernel.name == "redox" then [ redox_syscall_0_1_18 ] else [])
      ++ (if buildPlatform.parsed.kernel.name == "windows" then [ kernel32_sys_0_2_2 winapi_0_2_8 ] else []);
  };
```

I would recommend an example that kicks off the build by specifying the "root" Rust project only.

e.g.  `nix-build -E 'let pkgs = import <nixpkgs> {}; in (import ./hello.nix { inherit pkgs }).hello_0_1_0'`

I think that example expression will work.  Corrections and improvements welcome. :/

Anyway, it'll be subject to change if @Mic92's recommendation to replace `{pkgs}` to improve composability is applied.